### DISCUSSION TITLE
How to query supabase with multiple inputs/join tables

### DISCUSSION THREAD

**Discussion by MincePie** [supabase/supabase#17829] [Questions] [answered]
Is there any advice anywhere for how to compose a supabase query that will take several search inputs (free text search parameters, region (id on the main table), categories (each in a separate table, connected by a join table)? I have 6 join tables, a free text search input and two columns on the main table - any or all of them may be completed and I want to write a search query that searches using all of the completed fields.

I can see supabase has documentation showing how to do a single query but im looking to combine all the parameters and then do a query (with whichever and as many of the inputs as have been submitted).

**Comment by GaryAustin1** [comment, marked as answer]
You can add your filters by conditional adding them to the query.  You can build your select string as needed.
![image](https://github.com/supabase/supabase/assets/54564956/8b994601-3d71-4f0c-91f3-5169d720cfab)
https://supabase.com/docs/reference/javascript/using-filters
